packages/nc-gui/components/project/spreadsheet/components/MoreActions.vue

```diff
@@ -220,6 +237,58 @@ export default {
         })
       );
     },
+    async exportXlsx() {
+      let offset = 0
+      let c = 1
+      try {
+        while (!isNaN(offset) && offset > -1) {
+          let res;
+          if (this.publicViewId) {
+            res = await this.$api.public.csvExport(this.publicViewId, ExportTypes.CSV_DATA, {
+              query: {
+                fields:
+                  this.queryParams &&
+                  this.queryParams.fieldsOrder &&
+                  this.queryParams.fieldsOrder.filter(c => this.queryParams.showFields[c]),
+                offset,
+                sortArrJson: JSON.stringify(
+                  this.reqPayload &&
+                    this.reqPayload.sorts &&
+                    this.reqPayload.sorts.map(({ fk_column_id, direction }) => ({
+                      direction,
+                      fk_column_id,
+                    }))
+                ),
+                filterArrJson: JSON.stringify(this.reqPayload && this.reqPayload.filters),
+              },
+              headers: {
+                'xc-password': this.reqPayload && this.reqPayload.password,
+              },
+            });
+          } else {
+            res = await this.$api.dbViewRow.export(
+              'noco',
+              this.projectName,
+              this.meta.title,
+              this.selectedView.title,
+              ExportTypes.CSV_DATA
+            )
+            const { data } = res
+            const xlsx = XLSX.read(data.csvData, { type: 'string' })
```

Comment:
Seems like you can use `raw: true` to avoid unwanted processing of dates, also looks like it fixes extra empty cells.